test/cli/test-other.py

```diff
@@ -859,3 +860,69 @@ def test_build_dir_j_memleak(tmpdir): #12111
     ]
 
     assert_cppcheck(args, ec_exp=0, err_exp=[], out_exp=out_lines)
+
+
+def test_premium_with_relative_path(tmpdir):
+    product_name = 'Cppcheck Premium ' + str(time.time())
+
+    test_file = os.path.join(tmpdir, 'test.cpp')
+    with open(test_file, 'wt') as f:
+        f.write('int main() {}')
+
+    test_cfg = tmpdir.join('cppcheck.cfg')
+    with open(test_cfg, 'wt') as f:
+        f.write("""
+                {
+                    "addons": [],
+                    "productName": "NAME",
+                      "about": "Cppcheck Premium 1.2.3.4"
+                }
+                """.replace('NAME', product_name))
+
+    args = ['--premium=misra-c++-2008', test_file]
+
+
+    cppcheck_exe = copy_and_prepare_cppcheck(tmpdir)
+
+    args.insert(0, cppcheck_exe)
+
+    exitcode, _, stderr = cppcheck(args)
+    assert stderr == ''
+    assert exitcode == 0
+
+
+    exitcode, stdout, stderr = cppcheck([cppcheck_exe, '--version'])
+    assert stdout == product_name + '\n'
+    assert stderr == ''
+    assert exitcode == 0
+
+    os.remove(test_cfg)
```

Comment:
hmm an option would be to run without cppcheck.cfg first. doesn't really matter but I suggest that.